backend/src/controllers/manager/assignEmployeeCourses.ts

```diff
@@ -0,0 +1,77 @@
+import { NextFunction, Request, Response } from 'express';
+
+import BadRequestError from 'classes/errors/clientErrors/BadRequestError';
+import { INITIAL_INDX } from 'config/constants';
+import {
+  assignCourseToEmployee,
+  getAllClientCoursesProvider,
+} from 'db/providers/clientCourseProvider';
+import { materialsCounterProvider } from 'db/providers/courseProvider';
+import { getUserProvider } from 'db/providers/userProvider';
+import { ICourseToAssign } from 'interfaces/ICourses/IQueryCourses';
+import { generateProgressDto } from 'utils/dto/dtoUtils';
+import { checkCourseDuplicates } from 'utils/validation/checkDuplicates';
+import { removeCoursesToAssignDuplicates } from 'utils/normaliser/queryCourses';
+
+const assignEmployeeCourses = async (
+  req: Request<{ id: string }, never, ICourseToAssign[]>,
+  res: Response<never, { id: string; results: string }>,
+  next: NextFunction,
+) => {
+  try {
+    const { id: employeeId } = req.params;
+    const { id: managerId } = res.locals;
+    const coursesToAssign = req.body;
+
+    const employee = await getUserProvider(employeeId);
+    const manager = await getUserProvider(managerId);
+
+    if (String(employee.managerId) !== String(manager._id)) {
+      throw new BadRequestError('Employee does not belong to user.');
+    }
+
+    const reducedCoursesToAssign = removeCoursesToAssignDuplicates(coursesToAssign);
```

Comment:
It's pretty easy. Just pass as body something like
```
[
  {
     "courseId": "sameId"
  },
  {
     "courseId": "sameId"
  }
]
```

Reply:
but if i understand we get courses from our another query for get courses and there are no duplicates

Reply:
We're getting this courses from frontend, not from our provider or API endpoint. The situation when frontend accidentally trying to assign duplicated courses is quite possible IMHO. And I think, that data validation is still backend responsibility area

Reply:
Nobody canceled the "protection from the fool" 🤣

Reply:
otherwise i think it's quite hard to achive cause in frontend before they send body they get array of our unique courses then formed set of courses and will send it to us
your scenario can be achived only in manual mode for example if we try use postman

Reply:
quite hard doesn't mean impossible 😸 
I don't think we want to have our database be broken just because of some bugs on frontend during the process of development frontend part of the feature 😕

Reply:
and of course, is it really good API if we can broke down system just by sending request directly not using frontend application?